cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -1076,6 +1099,37 @@ private RetryableProvisionException handleApiException(ApiException e) throws Re
     throw new DataprocRuntimeException(e);
   }
 
+  //Throws retryable Exception for the cases that are transient in nature
+  private static void handleRetryableExceptions(Exception e) throws RetryableProvisionException {
+    // if there was an SocketTimeoutException ( read time out ) , we can just try again
+    if (e instanceof SocketTimeoutException) {
+      throw new RetryableProvisionException(e);
+    }
+
+    //Attempt retry in case of rate limit errors and service unavailable
+    if (e instanceof GoogleJsonResponseException) {
+      GoogleJsonResponseException gError = ((GoogleJsonResponseException) e);
+      int statusCode = gError.getStatusCode();
+
+      if (statusCode == 503) {
```

Comment:
Please use proper constants for status codes, e.g. `java.net.HttpURLConnection#HTTP_UNAVAILABLE` here.